lisa-kernel/src/main/scala/lisa/kernel/proof/SCProofChecker.scala

```diff
@@ -34,7 +34,7 @@ object SCProofChecker {
            *    Γ |- Δ
            */
           case Restate(s, t1) =>
-            if (isSameSequent(s, ref(t1))) SCValidProof(SCProof(step)) else SCInvalidProof(SCProof(step), Nil, s"The premise and the conclusion are not trivially equivalent.")
+            if (isImplyingSequent(ref(t1), s)) SCValidProof(SCProof(step)) else SCInvalidProof(SCProof(step), Nil, s"The premise does not trivially imply the conclusion.")
```

Comment:
It might be better to leave `Restate` as using `isSame`, as it ensures you do not accidentally miss a clause and lose information needed later in the proof. `Weakening` using `isImplying` makes intuitive sense, that way.